Framework/API/inc/MantidAPI/IMDHistoWorkspace.h

```diff
@@ -44,6 +44,10 @@ class DLLExport IMDHistoWorkspace : public IMDWorkspace,
   IMDHistoWorkspace_uptr clone() const {
     return IMDHistoWorkspace_uptr(doClone());
```

Comment:
At the public interface you are creating a `IMDHistoWorkspace_uptr`. You could just make this method virtual and do away with the `doClone`. As I say this is a minor point.